### PR TITLE
Supporting empty Content-Length headers by ignoring them.

### LINKED ISSUE #72: Error in getBodyAsString when content-length is set as a string.
_From @memmaker on December 25, 2016 22:2_

The current release version of SabreDAV bundles a faulty version of sabre-http (4.2.1).

If PHP is set to print out warnings, this release of sabre-http produces warnings in the output if a content-length header has been set, thus rendering the XML output invalid:

`stream_get_contents() expects parameter 2 to be integer, string given in /code/vendor/sabre/http/lib/Message.php on line 81`

Current versions of sabre-http have fixed this bug by casting the content-length to int.

If we could have the current composer file of SabreDAV pointing to a fixed release, this would not be an issue anymore.

_Copied from original issue: fruux/sabre-dav#926_

### PR DESCRIPTION
**Pull request by evert**
Fixes #72

### CODE REVIEW

File: lib/Message.php [hunk 1]
```diff
@@ -75,12 +75,11 @@ function getBodyAsString() {
             return '';
         }
         $contentLength = $this->getHeader('Content-Length');
-        if (null === $contentLength) {
-            return stream_get_contents($body);
-        } else {
+        if (is_int($contentLength) || ctype_digit($contentLength)) {
```
Comment (staabm) [inline lib/Message.php]: Does this lib already depend on the ctype ext?

Comment (staabm) [inline lib/Message.php]: Alternativ could be a cast to `int`

Comment (evert) [inline lib/Message.php]: It does. I didn't realize it was an extension. I'll add it to the `require` list to make sure this is captured

Comment (evert) [inline lib/Message.php]: The cast to int would turn an empty string into a 0, and instead I want an invalid value to be ignored altogether.

Comment (staabm) [inline lib/Message.php]:
```
$i = (int) $contentLength;
If ($i > 0 && $i != $contentLength)
...
```

But as you already depend on ctypes its fine.

Comment (evert) [inline lib/Message.php]: Ah not a bad one =)

**Pull request merged**